contracts/EasyAuction.sol

```diff
@@ -329,26 +329,20 @@ contract EasyAuction is Ownable {
 
         for (uint256 i = 0; i < iterationSteps; i++) {
             iterOrder = sellOrders[auctionId].next(iterOrder);
-            (, , uint96 sellAmountOfIter) = iterOrder.decodeOrder();
+            (, uint96 buyAmountOfIter, uint96 sellAmountOfIter) =
+                iterOrder.decodeOrder();
             sumBidAmount = sumBidAmount.add(sellAmountOfIter);
+            if (
+                sumBidAmount.mul(buyAmountOfIter) >=
+                auctioneerSellAmount.mul(sellAmountOfIter) ||
+                iterOrder == IterableOrderedOrderSet.QUEUE_END
+            ) {
+                sumBidAmount = sumBidAmount.sub(sellAmountOfIter);
+                iterOrder = sellOrders[auctionId].previous(iterOrder);
+                break;
+            }
         }
```

Comment:
This is my proposal without using `.previous()`. The reason I don't like `previous` is that it behaves differently from `.next()`, since it exists for deleted orders. This should not be an issue here, since we are on the main chain (this is enforced by the first call to `.next()`). However, I'd still try to avoid `previous` in case future changes break this assumption.

```
        bytes32 previousOrder;
        for (uint256 i = 0; i < iterationSteps; i++) {
            previousOrder = iterOrder;
            iterOrder = sellOrders[auctionId].next(iterOrder);
            (, uint96 buyAmountOfIter, uint96 sellAmountOfIter) =
                iterOrder.decodeOrder();
            sumBidAmount = sumBidAmount.add(sellAmountOfIter);
            if (
                sumBidAmount.mul(buyAmountOfIter) >=
                auctioneerSellAmount.mul(sellAmountOfIter) ||
                iterOrder == IterableOrderedOrderSet.QUEUE_END
            ) {
                sumBidAmount = sumBidAmount.sub(sellAmountOfIter);
                iterOrder = previousOrder;
                break;
            }
        }
```

Reply:
Nice, lets take this solution. 
I was under the impression that using a previousOrder might somehow increase the complexity. But for sure it does not. Good!